provider/k8s/metric_scraper.go

```diff
@@ -0,0 +1,154 @@
+package k8s
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/convox/convox/pkg/structs"
+	"github.com/pkg/errors"
+	am "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+type MetricScraperClient struct {
+	c       *http.Client
+	cluster kubernetes.Interface
+	host    string
+}
+
+func NewMetricScraperClient(cluster kubernetes.Interface, host string) *MetricScraperClient {
+	return &MetricScraperClient{
+		c: &http.Client{
+			Timeout: 10 * time.Second,
+		},
+		cluster: cluster,
+		host:    host,
+	}
+}
+
+func (m *MetricScraperClient) GetRackMetrics(opts structs.MetricsOptions) (structs.Metrics, error) {
+	if m.host == "" {
+		return nil, errors.WithStack(fmt.Errorf("unimplemented"))
+	}
```

Comment:
I kept it that way because console3 use this error for decision
see this comment for ref
https://github.com/convox/convox/pull/450#discussion_r916139404